### ISSUE TITLE
Graded ring

### ISSUE THREAD

**Issue by jvoight** [LMFDB/lmfdb#2627]
It would be nice if on the page associated to a congruence subgroup we listed a presentation for the graded ring of modular forms for that group, with generators (preferably eigenforms, so they can be clicked on) and relations among them.

**Comment by assaferan** [comment]
@jvoight - I think the relevant page is now part of Modular curves, e.g. in [this page](https://beta.lmfdb.org/ModularCurve/Q/35.48.3.a.1/) the generators are written as the newforms in the Jacobian, and the relations appear as the embedded model.
Does that settle this issue?

**Comment by jvoight** [comment]
No, unfortunately it does not.  The graded ring of modular forms is not always generated in weight 2, even when the canonical map gives a closed embedding into projective space.  It is when the stack is a scheme (so no elliptic points), like for [35.48.3.a.1](https://beta.lmfdb.org/ModularCurve/Q/35.48.3.a.1/).  

The content available in the full graded ring of modular forms is that you can get arbitrary weight k by multiplying forms.  It is also a description of the stacky curve in weighted projective space, which isn't the same as the coarse space.

It's fine to leave this as v2.0.

**Comment by assaferan** [comment]
I agree that with your observation that the graded ring is not the same as the canonical embedding, even when the canonical map is an embedding. However, as you said - it is a description of the (stacky) modular curve.
Therefore, I think that:
1. This is an issue for the Modular curves pages (and not for CMF)
2. Some decisions need to be made about the cases in which we store/display these, and I would like to hear what @roed314 thinks.

It's fine to leave for v2.0, but as we have all the ingredients to compute these, we may want to try for v1.4.

**Comment by jvoight** [comment]
Good point, I removed the CMF label!
